docs/source/developer-guide/perf-benchmarking.md

```
@@ -0,0 +1,468 @@
+In order to prepare a synthetic dataset, you can use the provided script in the `benchmarks/cpp`
+directory. For example, to generate a synthetic dataset of 1000 requests with a uniform ISL/OSL of
+128/128 for [meta-llama/Llama-3.1-8B](https://huggingface.co/meta-llama/Llama-3.1-8B), run:
+
+```shell
+python benchmarks/cpp/prepare_dataset.py --stdout --tokenizer meta-llama/Llama-3.1-8B token-norm-dist --input-mean 128 --output-mean 128 --input-stdev 0 --output-stdev 0 --num-requests 1000 > /tmp/synthetic_128_128.txt
+```
+
+### Running with the PyTorch Workflow
+
+To benchmark the PyTorch backend (`tensorrt_llm._torch`), use the following command with [dataset](#preparing-a-dataset) generated from previous steps. The `throughput` benchmark initializes the backend by tuning against the
+dataset provided via `--dataset` (or the other build mode settings described [above](#other-build-modes)).
```

markdownlint-cli2 (0.17.2)

Link fragments should be valid

(MD051, link-fragments)